fire/inspectutils.py

```diff
@@ -165,6 +166,10 @@ def Py3GetFullArgSpec(fn):
       varkw = name
     if param.annotation is not param.empty:
       annotations[name] = param.annotation
+      if "Union" in str(annotations[name]):
```

Comment:
This works for the example from #329 in Python 3.8, but

1. I think this will unfortunately not work for other types of `Union`s (e.g. `Union[str, int]` instead of `Option[int] == Union[int, None]`); and
2. it doesn't work in Python 3.9:

```
~/tmp ∃ python3 -m venv env
~/tmp ∃ source env/bin/activate
(env) ~/tmp ∃ pip install -U -q pip fire
(env) ~/tmp ∃ python --version
Python 3.9.9
(env) ~/tmp ∃ pip freeze
fire==0.4.0
six==1.16.0
termcolor==1.1.0
(env) ~/tmp ∃ cat test.py
#!/usr/bin/env python
from typing import Optional
from fire import Fire

def main(n: Optional[int] = None):
    n = 5 if n is None else n
    print(f"You chose {n}.")

if __name__ == "__main__":
    Fire(main)%
(env) ~/tmp ∃ ./test.py --help
INFO: Showing help with the command 'test.py -- --help'.

NAME
    test.py

SYNOPSIS
    test.py <flags>

FLAGS
    --n=N
        Type: Optional[typing.Optional[int]]
        Default: None
```